In [113]:
# Import necessary packages
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats as stats

#set up a large font for our pyplot
font = {'family' : 'Arial',
        'weight' : 'bold',
        'size'   : 22}

plt.rc('font', **font)

# Python's default exception mode is too verbose, let's set it to 'plain'
%xmode plain


Exception reporting mode: Plain


### Removing null values from dataframe

In [114]:
data = pd.read_csv("salary.csv")
data = data.drop(data.index[208])
data = data.sort_values('yearsworked')
data.tail()

salary  exprior  yearsworked  yearsrank  market  degree  otherqual  \
326  53686.0        0           31         17    0.78       1          0   
365  82508.0        0           32         23    0.93       0          1   
376  72152.0        0           33         22    1.00       1          0   
364  66334.0        0           35         23    0.78       1          0   
382  64109.0        0           41         28    0.91       1          0   

     position  male  Field  yearsabs  
326         3     1      1         1  
365         3     1      2         0  
376         3     1      2         2  
364         3     1      1         0  
382         3     1      2         0

### a)	Run a simple linear regression for Salary with one predictor variable: Years Worked

In [146]:
x = data['yearsworked'] # Years worked (independent variable) as x-axis
y = data['salary'] # Salary (dependent variable) as y-axis

# Calculate slope, intercept, r-value, p-value, and standard error using scipy
slope, intercept, r_value, p_value, std_err = stats.linregress(x, y)

# Calculate standard deviation of x and y
x_std = stats.tstd(x)
y_std = stats.tstd(y)

# Output all values
output  = "slope | intercept | r_value | p_value | x_std | y_std    | std_err\n"
output += '{:.2f}'.format(slope) + '|' + '{:9.2f}'.format(intercept) + '{:>3}'.format('|') + '{:5.2f}'.format(r_value)
output += '{:>5}'.format('|') + '{:5.2f}'.format(p_value) + '{:>5}'.format('|') + '{:5.2f}'.format(x_std)
output += '{:>3}'.format('|') + '{:9.2f}'.format(y_std) + '{:>2}'.format('|') + '{:6.2f}'.format(std_err)
print(output)

slope | intercept | r_value | p_value | x_std | y_std    | std_err
837.33| 40115.01  | 0.62    | 0.00    | 9.45  | 12685.13 | 46.44


### i.	What percentage of the variance in employees’ salaries is accounted for by the number of years they have worked?

In [149]:
x_var = stats.variation(data['yearsworked'])
x_var * 100

73.52476737919054

In [151]:
y_var = stats.variation(data['salary'])
y_var * 100

24.915376230136417

### >>> The variation of years worked is 73.52%, and the variation of salary is 24.92%.

### ii.	Does the model significantly predict the dependent variable? Report the amount of variance explained (R2) and significance value (p) to support your answer.

In [153]:
x_var ** 2

0.540589141816408